src/auth/authRepository.ts

```diff
@@ -34,63 +45,93 @@ export class AuthRepository implements IAuthRepository {
               `,
       [userName, userSurname, userEmail, userPassword]
     );
-
-    const notification = await this.database.query(`
-                insert into notifications ( user_id, code, type_of_notice)
-                VALUES ($1, $2, 'EMAIL_VERIFICATION');
-                `, [user.rows[0].id, code ]);
+    const expires_at = new Date(Date.now() + 15 * 60 * 1000);
+    await this.database.query(
+      `
+                insert into notifications (user_id, code, type_of_notice, expires_at)
+                VALUES ($1, $2, 'EMAIL_VERIFICATION', $3);
+                `,
+      [user.rows[0].id, code, expires_at]
+    );
 
     return user.rows[0];
   }
 
-
-  async isVerified(user_id: number){
-    const isVerified = await this.database.query(`
+  async addEmailCode(code: number, user_id: number){
+    const expires_at = new Date(Date.now() + 15 * 60 * 1000);
+    return this.database.query(
+      `
+                insert into notifications (user_id, code, type_of_notice, expires_at)
+                VALUES ($1, $2, 'EMAIL_VERIFICATION', $3);
+                `,
+      [user_id, code, expires_at]
+    );
+  }
+  async isVerified(user_id: number) {
+    const isVerified = await this.database.query(
+      `
       SELECT *
       from users
       where id = $1;
-      `, [user_id]);
-      return isVerified.rows[0].is_email_verified;
+      `,
+      [user_id]
+    );
+    return isVerified.rows[0].is_email_verified;
   }
 
-  async getCode(user_id: number){
-    const getCode = await this.database.query(`
+  async getCode(user_id: number) {
+    const getCode = await this.database.query(
+      `
      SELECT *
       from notifications
       where user_id = $1;
-      `, [user_id]);
-      return getCode.rows[0].code;
+      `,
+      [user_id]
+    );
+    return getCode.rows[0];
   }
 
-  async changeEmailIsVerified(user_id: number){
-    return this.database.query(`
+  async changeEmailIsVerified(user_id: number) {
+    return this.database.query(
+      `
       UPDATE users
         SET is_email_verified = true
         WHERE id = $1;
-        `, [user_id])
+        `,
+      [user_id]
+    );
   }
 
-  async addCode(code: number, id: number){
-    return this.database.query(`
+  async addCode(code: number, id: number) {
+    return this.database.query(
+      `
       insert into notifications (user_id, code, type_of_notice)
```

Comment:
same

Reply:
why you has two functions that creating notifiaction?
addCode and addEmailCode, you can use only one for every use case